js/m.viewer.js

```diff
@@ -521,22 +522,36 @@ function toggleAllLandmark()
   show_landmark = !show_landmark;
   var _list=document.getElementsByName("landmark");
   if(show_landmark) {
-    $('#landmarkbtn').addClass('pick');
-    for (var i=0; i<_list.length;i++) {
-      var _g=(_list[i].id).split('_').shift();
-      var _i=_list[i].value-1;
-      _list[i].checked=true;
-      landmarklist[_g][_i].visible=true;
-    }
+      $('#landmarkbtn').addClass('pick');
+      setLandmarksActive(true)
     } else {
       $('#landmarkbtn').removeClass('pick');
-      for (var i=0; i<_list.length;i++) {
-        var _g=(_list[i].id).split('_').shift();
-        var _i=_list[i].value-1;
-        _list[i].checked=false;
-        landmarklist[_g][_i].visible=false;
-      }
+      setLandmarksActive(false)
   }
+  updateCalculateDistanceButton()
+}
+
+// Enable or disable all landmarks, set to true or false based on the 'state'
+// parameter. Optionally, filterFunction can be passed in to choose which
+// landmarks are set to the new state.
+// filterFunction Example setting only landmarks for 'myMeshID':
+// setLandmarksActive(true, function(landmarkGroup, landmarkID) {return landmarkGroup===myMeshID;})
+function setLandmarksActive(state, filterFunction=null) {
+
+  var _list=document.getElementsByName("landmark");
+  for (var i=0; i<_list.length;i++) {
+    var _g=(_list[i].id).split('_').shift();
+    var _i=_list[i].value-1;
+    if (filterFunction == null || filterFunction(_g, _i) == true) {
+      _list[i].checked=state;
+      landmarklist[_g][_i].visible=state;
+    }
+  }
+}
```

Comment:
This process for looping over landmarks in the UI is copied in multiple locations. I consolidated a couple of them into one function above. Without a major refactor, I don't think there is a better way to access both landmarks in the UI and the list of landmarks tracked by XTK. One other addition is I added a filter function. The filter function is, in effect, a replacement for a `getLandmarksByMesh()` function, since that function would need to return both info about the UI state and the XTK state which are tracked in separate objects.